django_cloud_deploy/tool_requirements.py

```diff
@@ -0,0 +1,270 @@
+        if not shutil.which('kubectl'):
+            msg = 'To install kubectl run: gcloud components install kubectl'
+            raise MissingRequirementError(cls.NAME, msg)
+
+
+class CloudSqlProxy(Requirement):
+    NAME = 'Cloud Sql Proxy'
+
+    @classmethod
+    def check(cls):
+        """Checks if the requirement is installed.
+
+        Raises:
+            MissingRequirementError: If the requirement is not found.
+        """
+        if not shutil.which('cloud_sql_proxy'):
+            dl_link = 'https://cloud.google.com/sql/docs/mysql/sql-proxy'
+            msg = 'Please download Cloud SQL Proxy from {}'.format(dl_link)
+            raise MissingRequirementError(cls.NAME, msg)
+
+    @classmethod
+    def handle(cls, console: io.IO):
+        """Attempts to install the requirement.
+
+        Raises:
+            UnableToAutomaticallyInstall: If the installation fails.
+            NotImplementedError: If we expect the user to manually install the
+            requirement.
+        """
+        can_download = cls._prompt_download_requirement(console)
+        if can_download:
+            cls._download_cloud_sql_proxy(console)
+        else:
+            raise UnableToAutomaticallyInstall
+
+    @classmethod
+    def _prompt_download_requirement(cls, console: io.IO) -> bool:
+        while True:
+            console.tell('{} is required. y/n to download requirements'
+                         .format(cls.NAME))
+            default_answer = 'y'
+            answer = console.ask('[{}]: '.format(default_answer))
+
+            if not answer.strip() or answer.strip().lower() == 'y':
+                return True
+            elif answer.strip().lower() == 'n':
+                return False
+
+    @classmethod
+    def _download_cloud_sql_proxy(cls, console: io.IO):
+        import certifi
+
+        # We will install the proxy where gcloud is
+        gcloud_dir = os.path.dirname(shutil.which('gcloud'))
+        if sys.platform.startswith('linux'):
+            url = 'https://dl.google.com/cloudsql/cloud_sql_proxy.linux.amd64'
+        elif sys.platform.startswith('darwin'):
+            url = 'https://dl.google.com/cloudsql/cloud_sql_proxy.darwin.amd64'
+        else:
+            msg = '{} is not supported, only linux/mac.'.format(sys.platform)
+            raise NotImplementedError(msg)
+
+        console.tell('Downloading Cloud Sql Proxy')
+
+        file_name = 'cloud_sql_proxy'
+        executable_name = os.path.join(gcloud_dir, file_name)
+        with open(executable_name, 'wb') as file:
+            with urllib.request.urlopen(url, cafile=certifi.where()) as data:
+                file.write(data.read())
+
+        st = os.stat(executable_name)
+        os.chmod(executable_name, st.st_mode | stat.S_IRWXU)
+
+
+_REQUIREMENTS = [
+    Gcloud,
+    Docker,
+    Kubectl,
+    CloudSqlProxy
+]
+
+
+def check_and_handle_requirements(console: io.IO):
+    """Checks requirements are installs and attempts to install if possible.
```

Comment:
Checks that requirements are installed and attempts to install missing ones if possible.

Get rid of "if possible" if there is not enough space.